### PR TITLE
[risk=low][RW-6703] Some MailServiceImpl cleanup before adding the AAR emails

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Some minor refactoring to make it easier to implement RW-6703

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/mail/MailService.java [hunk 1]
```diff
@@ -14,9 +14,6 @@ void sendWelcomeEmail(final String contactEmail, final String password, final Us
   void sendInstitutionUserInstructions(final String contactEmail, final String userInstructions)
       throws MessagingException;
 
-  void sendBetaAccessCompleteEmail(final String contactEmail, final String username)
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailService.java:17]: no longer used.  There's one more beta access email that has a code reference.  That will be removed in RW-6913.

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 2]
```diff
@@ -86,40 +82,38 @@ public void sendBetaAccessRequestEmail(final String userName) throws MessagingEx
   public void sendWelcomeEmail(final String contactEmail, final String password, final User user)
       throws MessagingException {
 
-    final MandrillMessage msg =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:89]: Most sendEmail methods have a block like this.  Push this boilerplate into a new version of `sendWithRetries()`

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:89]: Please confirm that the parameters of these methods have not changed in your review.

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 3]
```diff
@@ -387,14 +349,6 @@ private String getRegistrationImage() {
     return cloudStorageClientProvider.get().getImageUrl("email_registration_example.png");
   }
 
-  // TODO choose desired date format
-  // currently will display '07/14/2020'
-  private String formatDate(final LocalDate expirationDate) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:392]: not used

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 4]
```diff
@@ -195,20 +176,6 @@ public void alertUserRegisteredTierExpiration(final DbUser user) {
...
-    final MandrillMessage msg =
-        new MandrillMessage()
-            .to(Collections.singletonList(validatedRecipient(contactEmail)))
-            .html(buildHtml(BETA_ACCESS_RESOURCE, betaAccessSubstitutionMap(username)))
-            .subject("All of Us ID Verification Complete")
-            .fromEmail(workbenchConfigProvider.get().mandrill.fromEmail);
-
-    sendWithRetries(msg, String.format("BetaAccess Complete for %s", contactEmail));
-  }
-
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:211]: I'm assuming this is no longer needed.

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 5]
```diff
@@ -131,25 +125,18 @@ public void alertUserFreeTierDollarThreshold(
...
-            .fromEmail(workbenchConfig.mandrill.fromEmail);
 
     sendWithRetries(
-        msg, String.format("User %s passed a free tier dollar threshold", user.getUsername()));
+        user.getContactEmail(),
+        String.format(
+            "Reminder - %s Free credit usage in All of Us Researcher Workbench",
+            formatPercentage(threshold)),
+        String.format("User %s passed a free tier dollar threshold", user.getUsername()),
+        msgHtml);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:135]: nit: change msgHtml to match expected arg name of htmlMessage like other calls to sendWithRetries

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 6]
```diff
@@ -158,17 +145,11 @@ public void alertUserFreeTierExpiration(final DbUser user) throws MessagingExcep
...
-            .html(msgHtml)
-            .subject(subject)
-            .fromEmail(workbenchConfig.mandrill.fromEmail);
-
     sendWithRetries(
-        msg, String.format("Free tier credits have expired for user %s", user.getUsername()));
+        user.getContactEmail(),
+        "Alert - Free credit expiration in All of Us Researcher Workbench",
+        String.format("Free tier credits have expired for user %s", user.getUsername()),
+        msgHtml);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:152]: nit: change msgHtml to match expected arg name of htmlMessage like other calls to sendWithRetries

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 7]
```diff
@@ -131,25 +125,18 @@ public void alertUserFreeTierDollarThreshold(
...
+        String.format("User %s passed a free tier dollar threshold", user.getUsername()),
+        msgHtml);
   }
 
   @Override
   public void alertUserFreeTierExpiration(final DbUser user) throws MessagingException {
-    final WorkbenchConfig workbenchConfig = workbenchConfigProvider.get();
 
     final String logMsg =
         String.format("Free credits have expired for User %s", user.getUsername());
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:142]: nit: make a String constant for this message and use it in the description arg as well when calling sendWithRetries

**Review by freemabd — APPROVED** [review]
Looks good.

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java (hunk 4 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:211]: correct

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java (hunk 7 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:142]: nice

**Pull request merged**